lib/scroll-through-time.js

```diff
@@ -69,32 +89,37 @@ export default {
     this.subscriptions.add(
       atom.config.observe('scroll-through-time.direction', (newValue) => {
         this.removeListener()
-        this.listener = getListener(newValue)
-        this.setupListener()
+        this.listener = getListener({direction: newValue, keyboard: this.getConfig('keyboardState')})
+        if (state.active) {
+          this.setupListener()
+        }
+      }));
+
+    this.subscriptions.add(
+      atom.config.observe('scroll-through-time.keyboardState', (newValue) => {
+          this.removeListener()
+          this.listener = getListener({direction: this.getConfig('direction'), keyboard: newValue})
+          if (state.active) {
+            this.setupListener()
+          }
       }));
 
-    if (this.getConfig('autoToggle') || state.active) {
-      this.toggle(true)
-    }
   },
 
   deactivate() {
-    this.active = false
     this.removeListener()
     this.subscriptions.dispose();
   },
 
   serialize() {
-    return {
-      active: this.active
-    };
+    return atom.config.get('scroll-through-time');
```

Comment:
hum if you do that, it means that it won't remember if it was active or not before closing atom. That's why there is `active` and `autoToggle`: it's not the same thing. `active` is the current state of activation while `autoToggle` means that it should be active on startup regardless of its previous state of activation

Reply:
I guess I'm confused about the deviation of active and autoToggle in the first place?  Shouldn't the user set config values be the source of truth for the state of the the package functionality?

I admit I'm ignorant since I'm not a long term user of the package. Is the current expected workflow that users only enable/disable (toggle) the feature as needed; using the provided commands to be efficient? If that's the case I think I'm starting to get where you're coming from.

So on that same train of thought, rather than having the modifier key check in the scroll handler I should simply make the key modifier toggle the feature on press and release. It seems more inline with the current functionality I suppose.

Reply:
Yes there are commands to activate or deactivate the scroll. And then there is the config which you set in the settings. One of the config is to activate the scroll on startup, regardless of the previous state.

 > I should simply make the key modifier toggle the feature on press and release

I think it's less efficient to set up the listener on each key press than having the listener and checking the key modifier.